src/hats_import/catalog/resume_plan.py

```diff
@@ -144,13 +145,12 @@ def get_remaining_map_keys(self):
         Returns:
             list of mapping keys *not* found in files like /resume/path/mapping_key.npz
         """
-        prefix = file_io.append_paths_to_pointer(self.tmp_path, self.HISTOGRAMS_DIR)
-        mapped_keys = self.get_keys_from_file_names(prefix, ".npz")
-        return [
-            (f"map_{i}", file_path)
-            for i, file_path in enumerate(self.input_paths)
-            if f"map_{i}" not in mapped_keys
+        prefix = file_io.get_upath(self.tmp_path) / self.HISTOGRAMS_DIR
+        done_indexes = [
+            int(re.match(r"map_(.*).npz", str(path.name)).group(1)) for path in prefix.glob("*.npz")
```

Comment:
```suggestion
            int(re.match(r"map_(\d+).npz", path.name).group(1)) for path in prefix.glob("*.npz")
```

Also: here and elsewhere, using `re.compile` to save patterns to module variables will improve performance further, perhaps enough to notice.